File: backend/app/schemas/batch.py
"""
批量发送相关的 Pydantic Schemas
"""
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from enum import Enum


class BatchStatus(str, Enum):
    """批量任务状态"""
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"
    CANCELLED = "cancelled"


class SmsBatchCreate(BaseModel):
    """创建批量任务"""
    batch_name: str = Field(..., min_length=1, max_length=200, description="批次名称")
    template_id: Optional[int] = Field(None, description="模板ID（使用模板时必填）")
    sender_id: Optional[str] = Field(None, max_length=20, description="发送方ID")
    send_config: Optional[Dict[str, Any]] = Field(default=None, description="发送配置")


class SmsBatchResponse(BaseModel):
    """批量任务响应"""
    id: int
    account_id: int
    batch_name: str
    template_id: Optional[int]
    file_path: Optional[str]
    file_size: Optional[int]
    total_count: int
    success_count: int
    failed_count: int
    processing_count: int
    status: BatchStatus
    error_message: Optional[str]
    sender_id: Optional[str]
    progress: int
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class SmsBatchListResponse(BaseModel):
    """批次列表响应"""
    total: int
    items: List[SmsBatchResponse]
    page: int
    page_size: int


class SmsBatchStats(BaseModel):
    """批次统计"""
    total_batches: int
    pending_batches: int
    processing_batches: int
    completed_batches: int
    failed_batches: int
    total_messages: int
    success_messages: int
    failed_messages: int


class BatchUploadResponse(BaseModel):
    """上传响应"""
    batch_id: int
    file_name: str
    file_size: int
    total_count: int
    message: str

File: frontend/src/api/batch.ts
import { request } from './index'

export interface SmsBatch {
  id: number
  account_id: number
  batch_name: string
  template_id: number | null
  file_path: string | null
  file_size: number | null
  total_count: number
  success_count: number
  /** 回执终态：已送达（SMSLog delivered） */
  delivered_count?: number
  /** 通道已接受但仍为 sent：终态回执未到或解析失败（与上游门户可能暂时不一致） */
  sent_awaiting_receipt_count?: number
  failed_count: number
  processing_count: number
  status: 'pending' | 'processing' | 'completed' | 'failed' | 'cancelled'
  error_message: string | null
  sender_id: string | null
  progress: number
  started_at: string | null
  completed_at: string | null
  created_at: string
  updated_at: string
}

// 上传批量发送文件
export const uploadBatchFile = (data: FormData) => {
  return request.post('/batches/upload', data, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

// 获取批次列表
export const getBatches = (params?: { page?: number; page_size?: number; status?: string }) => {
  return request.get('/batches', { params })
}

// 获取批次统计
export const getBatchStats = () => {
  return request.get('/batches/stats')
}

// 获取批次详情
export const getBatchDetail = (id: number) => {
  return request.get(`/batches/${id}`)
}

// 取消批次
export const cancelBatch = (id: number) => {
  return request.delete(`/batches/${id}`)
}

/** 失败记录重新计费并入队 */
export const retryBatchFailed = (id: number) => {
  return request.post<{ retried: number; skipped: number; errors: string[]; message: string }>(
    `/batches/${id}/retry-failed`
  )
}

/** 导出批次明细 CSV（后端手机号脱敏） */
export const exportBatchRecordsCsv = (id: number) => {
  return request.get(`/batches/${id}/export`, { responseType: 'blob' })
}
